src/strands/experimental/multiagent_hooks/multiagent_events.py

```diff
@@ -0,0 +1,55 @@
+"""Multi-agent execution lifecycle events for hook system integration.
+
+These events are fired by orchestrators (Graph/Swarm) at key points so
+hooks can persist, monitor, or debug execution. No intermediate state model
+is used—hooks read from the orchestrator directly.
+"""
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any
+
+from ...hooks.registry import BaseHookEvent
+
+if TYPE_CHECKING:
+    from ...multiagent.base import MultiAgentBase
+
+
+@dataclass
+class MultiAgentInitializationEvent(BaseHookEvent):
+    """Event triggered when multi-agent orchestrator initializes.
+
+    Attributes:
+        orchestrator: The multi-agent orchestrator instance
+        invocation_state: Configuration that user pass in
+    """
+
+    orchestrator: "MultiAgentBase"
+    invocation_state: dict[str, Any] | None = None
+
+
+@dataclass
+class AfterNodeInvocationEvent(BaseHookEvent):
```

Comment:
Per the [rules of hooks ](https://github.com/strands-agents/sdk-python/blob/99cd49bde350ddeba068f0567755e0fff651370e/src/strands/hooks/rules.md#hook-system-rules) for every *after* event, we should have a *before* event.